### PR TITLE
Icon state of turfs refactoring

### PR DESCRIPTION
**Pull request by UndeadNappist**
moves the handlers up for mineral floors up and genericizes them. Should fix a few issues, but creates others under unqiue conditions.

Note to self: Test before using - codebase and updates may have messed with things.

have been used in same codebase from previous vers. to fix various bugs.

## Changelog
refactor: refactored some code
/:cl:

### CODE REVIEW

**Comment by UndeadNappist** [comment]
Testing done. Works fine. Now with bonus commentary.

**Comment by UndeadNappist** [comment]
Works fine, tested fine. However, there's a runtime involved that is inherent to the master branch and not my changes.
[13:17:17] Runtime in atoms.dm,924: type mismatch: 0 |= the plating (122,125,2) (/turf/open/floor/plating)

Looks like someone screwed up typing with crowbars somewhere.

File: code/game/turfs/simulated/floor.dm [hunk 1]
```diff
@@ -21,6 +21,13 @@
 	var/floor_tile = null //tile that this floor drops
 	var/list/broken_states
 	var/list/burnt_states
+/*
```
Comment (Kmc2000) [inline code/game/turfs/simulated/floor.dm]:
```suggestion
/*Nsv13 - Turfs
```

File: code/game/turfs/simulated/floor.dm [hunk 2]
```diff
@@ -56,6 +63,8 @@
 		icon_regular_floor = icon_state
 	if(mapload && prob(33))
 		MakeDirty()
+	if (!icons)
```
Comment (Kmc2000) [inline code/game/turfs/simulated/floor.dm]:
```suggestion
	if(!islist(icons))
```

File: code/game/turfs/simulated/floor.dm (hunk 1 above)
Comment (Kmc2000) [reply code/game/turfs/simulated/floor.dm]: We do this to track changes across rebases more easily.

**Pull request merged**